### PR TITLE
add https connection and allowed hosts check

### PR DESCRIPTION
**Pull request by MattSilverio**
Resolving problem to issue #1553

### CODE REVIEW

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
The security issue highlighted concerns untrusted URL redirection from a remote source, where user input is incorporated into a URL redirect without full validation, leading to potential phishing attacks (CWE-601). The proposed change ensures that the scheme is HTTPS and the netloc is in the allowed hosts. However, this may not fully mitigate the risk as it does not address the validation of the entire URL, such as the path and query parameters which can also be exploited.

To address the issue more comprehensively, consider implementing the following:

1. Validate the full URL structure, not just the scheme and netloc.
2. Ensure that the path and query parameters do not redirect to an untrusted page even if the netloc is within the allowed hosts.
3. Use a list of allowed paths or a regular expression to validate the entire redirection endpoint.
4. Employ a safe redirect function that only redirects to known-safe URLs defined within the application.

Implementing these additional checks will help to ensure that the application is not vulnerable to open redirection attacks stemming from user input.

**Comment by JisanAR03** [comment]
@MattSilverio  the github , google and facebook callback function have to be updated. What you have to do is : read those social media documents and implement it on the BLT, which I'm going gonna do after few days, or if you can do that, implement this one, but don't do  something like sanitization system cause this is mainly feature problem, not a security one. best of luck ☺

**Review by MattSilverio — COMMENTED** [review]
views.py with additional checks as requested

**Comment by MattSilverio** [comment]
> The security issue highlighted concerns untrusted URL redirection from a remote source, where user input is incorporated into a URL redirect without full validation, leading to potential phishing attacks (CWE-601). The proposed change ensures that the scheme is HTTPS and the netloc is in the allowed hosts. However, this may not fully mitigate the risk as it does not address the validation of the entire URL, such as the path and query parameters which can also be exploited.
> 
> To address the issue more comprehensively, consider implementing the following:
> 
> 1. Validate the full URL structure, not just the scheme and netloc.
> 2. Ensure that the path and query parameters do not redirect to an untrusted page even if the netloc is within the allowed hosts.
> 3. Use a list of allowed paths or a regular expression to validate the entire redirection endpoint.
> 4. Employ a safe redirect function that only redirects to known-safe URLs defined within the application.
> 
> Implementing these additional checks will help to ensure that the application is not vulnerable to open redirection attacks stemming from user input.



> views.py with additional checks as requested

**Review by MattSilverio — COMMENTED** [review]
Additional checks implemented as requested, @DonnieBLT

**Review by MattSilverio — COMMENTED** [review]
Check last commits

File: .github/workflows/codeql.yml
Comment (DonnieBLT) [inline .github/workflows/codeql.yml:1]: we have a codeql file already I think you can remove this

**Review by DonnieBLT — COMMENTED** [review]
looks good, just let me know about the file

**Review by fredfalcon — APPROVED** [review]

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**